### PR TITLE
Resource reader added, and other refactors

### PR DESCRIPTION
**Pull request by bmozaffa**
New resource reader to help with listing and loading objects
Some refactor to resource writer and compare library

Signed-off-by: Babak Mozaffari <bmozaffa@redhat.com>

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -4,18 +4,15 @@
 
 [[constraint]]
   name = "sigs.k8s.io/controller-runtime"
-  # 0.2.0-beta.2 latest @dev time
-  version = "0.2.0-beta.2"
+  version = "v0.1.10"
```
Comment (bmozaffa) [inline Gopkg.toml:7]: @jeremyary we are basically unable to use this version in our operator due to conflicts with the operator-sdk version. I realize unit tests would not capture your platform detection functionality. Can you tell us if you really needed 0.2, and/or test and validate that everything still works?

Comment (jeremyary) [reply Gopkg.toml:7]: I checked this branch against OCP 3 and 4 clusters with no issue. I believe the version change was b/c operator-utils was using multiple overrides when I added this code, so it was a wholesale clean-up and update to all versions. 

Below is a dep status on my test operator to show I was using utils commit matching this PR and results against both clusters:
```
$ dep status
github.com/RHsyseng/operator-utils                   b7439e0 (override)                       b7439e0            4    

RESULTS AGAINST api.mw-ocp4.cloud.lab.eng.bos.redhat.com:6443
{"level":"info",...,"msg":"detecting platform version..."}
{"level":"info",...,"msg":"route.openshift.io found in apis, platform is OpenShift"}
{"level":"info",...,"msg":"PlatformInfo [Name: OpenShift, K8SVersion: 1.13+, OS: linux/amd64]"}
{"level":"info",...,"msg":"environment is openshift"}

RESULTS AGAINST master.middleware.ocp.cloud.lab.eng.bos.redhat.com:8443
{"level":"info",...,"msg":"detecting platform version..."}
{"level":"info",...,"msg":"route.openshift.io found in apis, platform is OpenShift"}
{"level":"info",...,"msg":"PlatformInfo [Name: OpenShift, K8SVersion: 1.11+, OS: linux/amd64]"}
{"level":"info",...,"msg":"environment is openshift"}
```

**Review by jeremyary — APPROVED** [review]

File: Gopkg.toml (hunk 1 above)
Comment (bmozaffa) [reply Gopkg.toml:7]: @jeremyary Great, thanks! However, I've found that dependency issues go beyond dep and compile time. You will often not see a problem till you try to use a function. Is that not the case here, and can you confirm that this PR doesn't break your functions?

Comment (jeremyary) [reply Gopkg.toml:7]: The results above are from calling operator-utils .IsOpenShift() (uses my code) from a running operator via sdk up local, so should cover runtime issues, no?

**Pull request merged**